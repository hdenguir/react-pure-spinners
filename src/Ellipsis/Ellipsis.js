import React from 'react'
import styles from './styles.css'

const Ellipsis = () => (
  <>
    <style>{`${styles}`}</style>
    <div className="lds-ellipsis">
      <div></div>
      <div></div>
      <div></div>
      <div></div>
    </div>
  </>
)

export default Ellipsis
