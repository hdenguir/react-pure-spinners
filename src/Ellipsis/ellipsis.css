:root {
  --rps-ellipsis-color: #ccc;
  --rps-ellipsis-size: 64px;
}

.rps-ellipsis {
  display: inline-block;
  position: relative;
  width: var(--rps-ellipsis-size);
  height: var(--rps-ellipsis-size);
}

.rps-ellipsis div {
  position: absolute;
  top: calc(var(--rps-ellipsis-size) / 2.42424);
  width: calc(var(--rps-ellipsis-size) / 6.15385);
  height: calc(var(--rps-ellipsis-size) / 6.15385);
  border-radius: 50%;
  background: var(--rps-ellipsis-color);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}

.rps-ellipsis div:nth-child(1) {
  left: calc(var(--rps-ellipsis-size) / 10);
  animation: rps-ellipsis1 0.6s infinite;
}

.rps-ellipsis div:nth-child(2) {
  left: calc(var(--rps-ellipsis-size) / 10);
  animation: rps-ellipsis2 0.6s infinite;
}

.rps-ellipsis div:nth-child(3) {
  left: calc(var(--rps-ellipsis-size) / 2.5);
  animation: rps-ellipsis2 0.6s infinite;
}

.rps-ellipsis div:nth-child(4) {
  left: calc(var(--rps-ellipsis-size) / 1.42857);
  animation: rps-ellipsis3 0.6s infinite;
}

@keyframes rps-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}

@keyframes rps-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}

@keyframes rps-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(calc(var(--rps-ellipsis-size) / 3.33333), 0);
  }
}
