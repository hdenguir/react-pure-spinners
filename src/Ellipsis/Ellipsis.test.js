import Ellipsis from './Ellipsis'

import React from 'react'

import renderer from 'react-test-renderer'

it('Redners Correctly', () => {
  const tree = renderer.create(<Ellipsis />).toJSON()
  expect(tree).toMatchSnapshot()
})
