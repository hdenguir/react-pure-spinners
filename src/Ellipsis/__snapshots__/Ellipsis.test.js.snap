// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redners Correctly 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="lds-ellipsis"
  >
    <div />
    <div />
    <div />
    <div />
  </div>,
]
`;
